src/components/confirm/confirm.pw.tsx

```diff
@@ -0,0 +1,648 @@
+      await expect(button).toHaveCSS("color", colour);
+      await expect(button).toHaveCSS("border-color", border);
+    });
+  });
+
+  cancelButtonTypes.forEach(([type, colour, border]) => {
+    test(`should check cancel button is ${type} type`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ConfirmComponent cancelButtonType={type} />);
+
+      const button = page.getByRole("button").filter({ hasText: "No" });
+      await expect(button).toHaveCSS("color", colour);
+      await expect(button).toHaveCSS("border-color", border);
+    });
+  });
+
+  ([
+    [true, "disabled"],
+    [false, "enabled"],
+  ] as [ConfirmProps["disableCancel"], string][]).forEach(
+    ([boolVal, state]) => {
+      test(`should check when disableCancel prop is ${boolVal} that cancel button is ${state}`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(<ConfirmComponent disableCancel={boolVal} />);
+
+        const button = page.getByRole("button").filter({ hasText: "No" });
+        if (boolVal) {
+          await expect(button).toHaveAttribute("disabled", "");
+          await expect(button).toBeDisabled();
+        } else {
+          await expect(button).not.toHaveAttribute("disabled", "");
+          await expect(button).toBeEnabled();
+        }
+      });
+    }
+  );
+
+  ([
+    [true, "disabled"],
+    [false, "enabled"],
+  ] as [ConfirmProps["disableConfirm"], string][]).forEach(
+    ([boolVal, state]) => {
+      test(`should check when disableConfirm prop is ${boolVal} that confirm button is ${state}`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(<ConfirmComponent disableConfirm={boolVal} />);
+
+        const button = page.getByRole("button").filter({ hasText: "Yes" });
+        if (boolVal) {
+          await expect(button).toHaveAttribute("disabled", "");
+          await expect(button).toBeDisabled();
+        } else {
+          await expect(button).not.toHaveAttribute("disabled", "");
+          await expect(button).toBeEnabled();
+        }
+      });
+    }
+  );
+
+  test(`should check confirm button is loading`, async ({ mount, page }) => {
+    await mount(<ConfirmComponent isLoadingConfirm />);
+
+    const button = getDataElementByValue(page, "confirm");
+    await expect(button).toHaveAttribute("disabled", "");
+    const loader = page.getByLabel("Loading");
+    await expect(loader).toBeAttached();
+  });
+
+  test(`should check Esc key is disabled`, async ({ mount, page }) => {
+    await mount(<ConfirmComponent disableEscKey />);
+
+    await checkDialogIsInDOM(page);
+    await page.keyboard.press("Escape");
+    await checkDialogIsInDOM(page);
+  });
+
+  test(`should check close icon is enabled`, async ({ mount, page }) => {
+    await mount(<ConfirmComponent showCloseIcon />);
+
+    const closeIcon = page.getByLabel("close");
+    await closeIcon.click();
+    await expect(page.getByRole("alertdialog")).not.toBeAttached();
+  });
+
+  ([
```

Comment:
suggestion: this array and the one used below could be stored in a const as `color` and `borderColor` are the same etc

```
 const destructiveState: [ConfirmProps["cancelButtonDestructive"], string][] = [
[true, "rgb(203, 55, 74)"],
 [false, "rgb(0, 126, 69)"]
 ];
 
 ...
 
 destrucitveStates..forEach(
    ([state, color]) => {
      test(`should check cancel button has destructive properties when cancelButtonDestructive prop is ${state}`, async ({
        mount,
        page,
      }) => {
        await mount(<ConfirmComponent cancelButtonDestructive={state} />);

        const button = page.getByRole("button").filter({ hasText: "No" });
        await expect(button).toHaveCSS("color", color);
        await expect(button).toHaveCSS("border-color", color);
      });
    }
  );